### PR TITLE
[Improvement] Refactor FlinkProcessRegistrar to be more extensible

### PR DESCRIPTION
**Pull request by mproch**
This is mostly code-shuffling. The aim is to make it easier to override default NK behaviour/configuration of Flink job, to make it easier to configure model to use older Flink version

### CODE REVIEW

File: engine/flink/generic/src/test/scala/pl/touk/nussknacker/genericmodel/GenericItSpec.scala [hunk 1]
```diff
@@ -321,14 +322,14 @@ class GenericItSpec extends FunSuite with FlinkSpec with Matchers with KafkaSpec
...
-  private var registrar: FlinkStreamingProcessRegistrar = _
+  private var registrar: FlinkProcessRegistrar = _
   private lazy val valueSerializer = new KafkaAvroSerializer(schemaRegistryMockClient)
   private lazy val valueDeserializer = new KafkaAvroDeserializer(schemaRegistryMockClient)
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
     val modelData = LocalModelData(config, creator)
-    registrar = FlinkStreamingProcessRegistrar(new FlinkProcessCompiler(modelData), config, ExecutionConfigPreparer.unOptimizedChain(modelData, None))
+    registrar = process.registrar.FlinkProcessRegistrar(new FlinkProcessCompiler(modelData), config, ExecutionConfigPreparer.unOptimizedChain(modelData, None))
```
Comment (lciolecki) [inline engine/flink/generic/src/test/scala/pl/touk/nussknacker/genericmodel/GenericItSpec.scala:332]: Why `process.registrar.` instead simply `FlinkProcessRegistrar`? :)

File: engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/runner/FlinkStreamingProcessMain.scala [hunk 2]
```diff
@@ -21,9 +22,15 @@ object FlinkStreamingProcessMain extends FlinkProcessMain[StreamExecutionEnviron
...
                                     prepareExecutionConfig: ExecutionConfigPreparer): Unit = {
     val compiler = new FlinkProcessCompiler(modelData)
-    val registrar = FlinkStreamingProcessRegistrar(compiler, modelData.processConfig, prepareExecutionConfig)
+    val registrar: FlinkProcessRegistrar = prepareRegistrar(modelData, prepareExecutionConfig, compiler)
     registrar.register(env, process, processVersion)
     val preparedName = modelData.objectNaming.prepareName(process.id, modelData.processConfig, new NamingContext(FlinkUsageKey))
     env.execute(preparedName)
   }
+
+  protected def prepareRegistrar(modelData: ModelData,
```
Comment (lciolecki) [inline engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/runner/FlinkStreamingProcessMain.scala:31]: Do we realy need this oneline method?

File: engine/flink/generic/src/test/scala/pl/touk/nussknacker/genericmodel/GenericItSpec.scala (hunk 1 above)
Comment (mproch) [reply engine/flink/generic/src/test/scala/pl/touk/nussknacker/genericmodel/GenericItSpec.scala]: it was some stupid auto refactor I think

File: engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/runner/FlinkStreamingProcessMain.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/runner/FlinkStreamingProcessMain.scala]: You're right, method above is small enough...

**Pull request merged**